completers/pulumi_completer/cmd/action/urn.go

```diff
@@ -1,60 +1,53 @@
 package action
 
 import (
-	"path/filepath"
-	"regexp"
-	"strings"
-	"time"
-
+	"encoding/json"
 	"github.com/rsteube/carapace"
 	"github.com/rsteube/carapace/pkg/cache"
 	"github.com/spf13/cobra"
+	"path/filepath"
+	"time"
 )
 
-type resource struct {
-	Urn string
-	Id  string
+type Stack struct {
+	Deployment struct {
+		Resources []struct {
+			Urn string `json:"urn"`
+		} `json:"resources"`
+	} `json:"deployment"`
 }
 
 func ActionUrns(cmd *cobra.Command) carapace.Action {
 	return carapace.ActionCallback(func(c carapace.Context) carapace.Action {
-		return carapace.ActionCallback(func(c carapace.Context) carapace.Action {
-			cwd := cmd.Flag("cwd").Value.String()
-			stack := cmd.Flag("stack").Value.String()
-
-			c.Setenv("PULUMI_SKIP_UPDATE_CHECK", "1")
-			return carapace.ActionExecCommand("pulumi", "--cwd", cwd, "stack", "--stack", stack, "--show-urns")(func(output []byte) carapace.Action {
-				reUrn := regexp.MustCompile(`URN: (?P<urn>urn:pulumi.*)`)
-				reId := regexp.MustCompile(`ID: (?P<id>.*)`)
-
-				resources := make([]resource, 0)
-				for _, line := range strings.Split(string(output), "\n") {
-					if reUrn.MatchString(line) {
-						resources = append(resources, resource{Urn: reUrn.FindStringSubmatch(line)[1]})
-					}
-					if reId.MatchString(line) {
-						resources[len(resources)-1].Id = reId.FindStringSubmatch(line)[1]
-					}
+		cwd := cmd.Flag("cwd").Value.String()
+		stack := cmd.Flag("stack").Value.String()
+
+		c.Setenv("PULUMI_SKIP_UPDATE_CHECK", "1")
+		return carapace.ActionExecCommand("pulumi", "stack", "export", "--cwd", cwd, "--stack", stack)(func(output []byte) carapace.Action {
+			var stack Stack
+			if err := json.Unmarshal(output, &stack); err != nil {
+				return carapace.ActionMessage(err.Error())
+			}
+			urns := make([]string, len(stack.Deployment.Resources))
+			for i, r := range stack.Deployment.Resources {
+				urns[i] = r.Urn
+			}
+			return carapace.ActionValues(urns...)
+		}).Cache(5*time.Second,
```

Comment:
I think we can remove the caching. Unless you experience slowness somewhere, but i don't see repetitive calls to it at the moment where the short cache would help :thinking:

Reply:
Just tested the cache again, and I think it is a good thing to have. In my case it usually takes 2 `<tab>`s to complete a `urn` (in `elvish). Each `pulumi stack export` takes ~ 2 seconds. I would say it is even safe to increase the cache timeout, as these are typically slow changing resources.

Reply:
Yeah, increase it to what you think makes sense (don't forget about newly created resources).

Reply:
Apologies, missed this comment. Thank you!